x/programs/examples/imports/pstate/pstate.go

```diff
@@ -260,3 +271,25 @@ func (i *Import) deleteFn(caller *program.Caller, memOffset int32, size int32) (
 
 	return types.ValI32(int32(ptr)), nil
 }
+
+func (i *Import) logFn(caller *program.Caller, memOffset int32, size int32) (*types.Val, error) {
+	memory, err := caller.Memory()
+	if err != nil {
+		i.log.Error("failed to get memory from caller",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+
+	bytes, err := memory.Range(uint32(memOffset), uint32(size))
+	if err != nil {
+		i.log.Error("failed to read args from program memory",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+
+	fmt.Fprintf(os.Stderr, "%s\n", bytes)
```

Comment:
We might want to log to `Stdout` instead of `Stderr` in the future but it would interfere with the simulator communication layer. This is to stick with Rust's std `dbg!` so that it could be not captured for successful tests with `--nocapture`